src/components/JobCard/__test__/JobCard.test.tsx

```diff
@@ -0,0 +1,101 @@
+import { describe, expect, it, vi } from 'vitest'
+import { render } from '@/tests/customRender'
+import { screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import JobCard from '../JobCard'
+
+// Mock data for the job
+const mockJob = {
+    title: 'Software Engineer',
+    company: 'Tech Corp',
+    location: 'Tokyo, Shibuya',
+    salary: '600,000 - 12,000,000',
+    tags: ['React', 'JavaScript', 'Frontend'],
+    logoUrl: 'https://via.placeholder.com/150',
+    publicationDate: '31/01/2025',
+    jobPostingUrl: 'https://jobpostingurl.com',
+}
+
+describe('JobCard Component', () => {
+    it('should render the job details correctly', () => {
+        render(<JobCard job={mockJob} />)
+
+        // Check the job title
+        expect(screen.getByText(mockJob.title)).toBeInTheDocument()
+
+        // Check the company name
+        expect(screen.getByText(mockJob.company)).toBeInTheDocument()
+
+        // Check the location
+        expect(screen.getByText(mockJob.location)).toBeInTheDocument()
+
+        // Check the salary
+        expect(screen.getByText(mockJob.salary)).toBeInTheDocument()
+
+        // Check the tags
+        mockJob.tags.forEach((tag) => {
+            expect(screen.getByText(tag)).toBeInTheDocument()
+        })
+
+        // Check the publication date
+        expect(screen.getAllByText(mockJob.publicationDate).length).toBeGreaterThan(0)
```

Comment:
Let's simplify the assertion by removing the char length check, and use async with retries

```suggestion
     const publicationDate = await screen.findByText(mockJob.publicationDate)
     expect(publicationDate).toBeVisible()
```

Reply:
Here I got this error:
```
should render the job details correctly
TestingLibraryElementError: Found multiple elements with the text: 2025/03/31
```

So I apply this as a solution:
```
 const publicationDate = await screen.findAllByText(mockJob.publicationDate)
        expect(publicationDate[0]).toBeVisible()
```
Let me know any thoughts

Reply:
The difference is using a command `FindAllBy...` returns an array. Please use the singular selector in the example above

Reply:
Pardon my earlier comment, I see you are finding multiple elements. 

Are there multiple identical dates in the DOM? Maybe we need a more specific selector to ensu an accurate test, like selecting by role. I'm not at my computer right now, but I can take a closer look later this evening.

Reply:
I'm looking at the code for `JobCard.tsx` and see the publication date and surrounding code written twice in the source. It seems one time is for the mobile view, and the other is for desktop viewports.

Let's write an assertion that two exist, and that only one of the two is visible. This will give us confidence that the page displays correctly. Alternatively, we could add a testid to distinguish the mobile and desktop versions, and then select the DOM element for one of the two, and then search for the string within that.